tests/foreman/cli/test_puppetmodule.py

```diff
@@ -1,117 +0,0 @@
-"""Test class for PuppetModule CLI
-
-:Requirement: Puppetmodule
-
-:CaseAutomation: Automated
-
-:CaseLevel: Acceptance
-
-:CaseComponent: Puppet
-
-:Assignee: vsedmik
-
-:TestType: Functional
-
-:CaseImportance: High
-
-:Upstream: No
-"""
-import pytest
-
-from robottelo.cli.factory import make_product
-from robottelo.cli.factory import make_repository
-from robottelo.cli.puppetmodule import PuppetModule
-from robottelo.cli.repository import Repository
-from robottelo.config import settings
-from robottelo.constants.repos import FAKE_0_PUPPET_REPO
-from robottelo.constants.repos import FAKE_1_PUPPET_REPO
-
-
-pytestmark = [
-    pytest.mark.skipif(
-        not settings.robottelo.REPOS_HOSTING_URL,
-        reason="repos_hosting_url is not defined in robottelo.properties",
-    )
-]
-
-
-@pytest.fixture(scope='module')
-def module_setup(module_org):
-    product = make_product({'organization-id': module_org.id})
-    repo = make_repository(
-        {
-            'organization-id': module_org.id,
-            'product-id': product['id'],
-            'content-type': 'puppet',
-            'url': FAKE_0_PUPPET_REPO,
-        }
-    )
-    Repository.synchronize({'id': repo['id']})
-    return {'org': module_org, 'product': product, 'repo': repo}
-
-
-@pytest.mark.tier1
-def test_positive_list(module_setup):
-    """Check if puppet-module list retrieves puppet-modules of
-    the given org
-
-    :id: 77635e70-19e7-424d-9c89-ec5dbe91de75
-
-    :expectedresults: Puppet-modules are retrieved for the given org
-
-    :BZ: 1283173
-
-    :CaseImportance: Critical
-    """
-    result = PuppetModule.list({'organization-id': module_setup['org'].id})
-    # There are 4 puppet modules in the test puppet-module url
-    assert len(result) == 4
-
-
-@pytest.mark.tier1
-def test_positive_info(module_setup):
-    """Check if puppet-module info retrieves info for the given
-    puppet-module id
-
-    :id: 8aaa9243-5e20-49d6-95ce-620cc1ba18dc
-
-    :expectedresults: The puppet-module info is retrieved
-
-    :CaseImportance: Critical
-    """
-    return_value = PuppetModule.list({'organization-id': module_setup['org'].id})
-    for i in range(len(return_value)):
-        result = PuppetModule.info({'id': return_value[i]['id']}, output_format='json')
-        assert result['id'] == return_value[i]['id']
-
-
-@pytest.mark.tier1
-@pytest.mark.upgrade
```

Comment:
I think puppet would be removed for newly installed satellite but if the customer has upgraded satellite then puppet will work as it is, so I would suggest to keep the upgrade marker test for verifying the behavior of puppet after the upgrade.

Reply:
@devendra104 makes sense. Probably we should also go through upgrade marker tests again to have better coverage.

Reply:
@devendra104 @jameerpathan111 I'll run through them and see what kind of results I get.  If some of the upgrade tests still uses puppet module for publishing, then we're have to readjust as necessary.

Reply:
I still think this can be removed. But I guess one of us needs to prove the correctness of his idea by manually upgrading the satellite

Reply:
@devendra104 ^^

Firstly 2 stubbed test cases can be removed. They are not going to be implemented.
And `api/test_puppetmodule` and `cli/test_puppetmodule` uses `Repository` entity, and repository does not support content_type `puppet` in sat 6.10.  And puppet modules api endpoints are no longer callable. 
So even the content is saved on the drive it doesn't matter, because we have no satellite way to test it.

Upgrade scenarious should be consired in other tests where we test puppet environments and puppet classes.

Reply:
yes, you are right @tstrych after the upgrade no puppet content would exist on the setup so we are good to remove the content-related test cases.